src/main/scala/io/iohk/ethereum/domain/branch/Branch.scala

```diff
@@ -2,17 +2,8 @@ package io.iohk.ethereum.domain.branch
 
 import akka.util.ByteString
 
-import io.iohk.ethereum.domain.Block
+sealed trait Branch
 
-/** An interface to manipulate blockchain branches */
-trait Branch {
+case class BestBranch(tipBlockHash: ByteString, tipBlockNumber: BigInt) extends Branch
```

Comment:
I can imagine this case class being used for all the branches, not only the best one

Reply:
It does not contain data specific to the best branch, yes. Right now it is more a marker type to tell the code that we can use the block number -> hash index.